components/governance/org.wso2.carbon.governance.lcm/src/main/java/org/wso2/carbon/governance/lcm/tasks/LCNotificationScheduler.java

```diff
@@ -0,0 +1,323 @@
+     * @param lifecycleName         lifecycle name.
+     * @param lifecycleState        new lifecycle state after the state changed.
+     * @param isInvokeAspect        is this add scheduler method is called when invoking a lifecycle.
+     * @throws GovernanceException  Throws when an error occurred while adding a lifecycle notification scheduler to
+     *                              the database.
+     */
+    public void addScheduler(ResourceImpl resource, String lifecycleName, String lifecycleState, boolean isInvokeAspect)
+            throws GovernanceException {
+        if (resource != null && StringUtils.isNotEmpty(lifecycleName) && StringUtils.isNotEmpty(lifecycleState)) {
+
+            List checkpointsList = LifecycleStateDurationUtils.getDurationBeans(lifecycleName, lifecycleState);
+            if (checkpointsList != null) {
+                // Iterate through the checkpoint objects.
+                for (Object checkpoint : checkpointsList) {
+                    OMElement checkpointElement = (OMElement) checkpoint;
+                    OMElement boundary = checkpointElement.getFirstChildWithName(
+                            new QName(LifecycleConstants.LIFECYCLE_CONFIGURATION_NAMESPACE_URI,
+                                    LifecycleConstants.LIFECYCLE_CHECKPOINT_BOUNDARY));
+                    String minTimestamp = boundary
+                            .getAttribute(new QName(LifecycleConstants.LIFECYCLE_LOWER_BOUNDARY)).getAttributeValue();
+                    String maxTimestamp = boundary
+                            .getAttribute(new QName(LifecycleConstants.LIFECYCLE_UPPER_BOUNDARY)).getAttributeValue();
+
+                    long durationDifference = getDurationDifference(minTimestamp, maxTimestamp);
+
+                    // Set scheduler property.
+                    String propertyValue = lifecycleName + "." + checkpointElement.getAttribute(new QName(checkpointId))
+                            .getAttributeValue() + "." + getNotificationDate(durationDifference);
+                    resource.addProperty(lcCheckPointPropertyName, propertyValue);
+                    // If the lifecycle is invoke registry.put is need to add the property because when invoke is
+                    // triggered registry put method is not called.
+                    if (isInvokeAspect) {
+                        try {
+                            CommonUtil.getRootSystemRegistry().put(resource.getPath(), resource);
+                        } catch (RegistryException e) {
+                            throw new GovernanceException(
+                                    "Error while saving resource when lifecycle is invoked after adding property '"
+                                            + propertyValue + "' to " + resource.getPath(), e);
+                        }
+                    }
+                }
+            } else {
+                if (log.isDebugEnabled()) {
```

Comment:
Hi,

Removed this else block because of the introduction of Validattion for Illegal arguments in pull request [1], commit[2].
[1] - https://github.com/wso2/carbon-governance/pull/44
[2] - https://github.com/prasa7/carbon-governance/commit/c681dda86c2df780a044e475a55cd5a4b2f5ef89

Thanks